lib/view_component/test_helpers.rb

```diff
@@ -126,6 +126,22 @@ def with_request_url(path)
       @controller = old_controller
     end
 
+    # Set the current_user in the given controller to the given value:
+    #
+    # ```ruby
+    # with_current_user(user) do
+    #  render_inline(MyComponent.new)
+    # end
+    # ```
+    #
+    # @param user [Object] The user to set as the current_user.
+    def with_current_user(user)
+      controller.define_singleton_method(:current_user) { user }
+      yield
+    ensure
+      controller.define_singleton_method(:current_user) { nil }
+    end
```

Comment:
I'm not convinced this is the best approach, but I'm haven't been able to think of other approaches yet that would work and don't depend on a particular testing framework.

Reply:
I feel like this is a great application-level helper, but I'm not sure how it fits into the framework. In the GitHub codebase we've defined our own custom method for this that is specific to our application.

 Do you know if/how authentication library helper methods like Devise's [`sign_in`](https://www.rubydoc.info/gems/devise/Devise/Test/ControllerHelpers) helper work with ViewComponent? I think that could provide some guidance on how to move forward here.